envoy/api/v2/auth/rbac.proto

```diff
@@ -0,0 +1,181 @@
+// (https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html). These attributes are
+// mapped to the attributes defined in AttributeContext
+// (https://github.com/envoyproxy/data-plane-api/blob/master/envoy/service/auth/v2/attribute_context.proto)
+// by the RBAC engine. Here are some examples of Istio attribute names:
+// source.service, source.labels["app"], request.headers["version"].
+//
+// Below is an example of ServiceRoleBinding object "test-binding-products", which binds two
+// subjects to ServiceRole "product-viewer":
+//   * User (service account) "cluster.local/ns/default/sa/bookinfo-reviews"
+//   * Service "reviews.default.svc.cluster.local" at version "v1".
+//
+//     apiVersion: "config.istio.io/v1alpha1"
+//     kind: ServiceRoleBinding
+//     metadata:
+//       name: test-binding-products
+//       namespace: default
+//     spec:
+//       subjects:
+//       - user: cluster.local/ns/default/sa/bookinfo-reviews
+//       - properties:
+//           source.service: "reviews.default.svc.cluster.local"
+//           source.labels["version"]: "v1"
+//       roleRef:
+//         kind: ServiceRole
+//         name: "products-viewer"
+
+// ServiceRole specification contains a list of access rules (permissions).
+// This represent the "Spec" part of the ServiceRole object. The name and namespace
+// of the ServiceRole is specified in "metadata" section of the ServiceRole object.
+message ServiceRole {
+  // Required. The set of access rules (permissions) that the role has.
+  repeated AccessRule rules = 1;
+}
+
+// AccessRule defines a permission to access a list of services.
+message AccessRule {
+  // Required. A list of service names.
+  // Exact match, prefix match, and suffix match are supported for service names.
+  // For example, the service name "bookstore.mtv.svc.cluster.local" matches
+  // "bookstore.mtv.svc.cluster.local" (exact match), or "bookstore*" (prefix match),
+  // or "*.mtv.svc.cluster.local" (suffix match).
+  // If set to ["*"], it refers to all services in the namespace.
+  repeated string services = 1;
+
+  // Optional. A list of HTTP paths or gRPC methods.
+  // gRPC methods must be presented as fully-qualified name in the form of
+  // packageName.serviceName/methodName.
+  // Exact match, prefix match, and suffix match are supported for paths.
+  // For example, the path "/books/review" matches
+  // "/books/review" (exact match), or "/books/*" (prefix match),
+  // or "*/review" (suffix match).
+  // If not specified, it applies to any path.
+  repeated string paths = 2;
+
+  // Optional. A list of HTTP methods (e.g., "GET", "POST").
+  // It is ignored in gRPC case because the value is always "POST".
+  // If set to ["*"] or not specified, it applies to any method.
+  repeated string methods = 3;
+
+  // Definition of a custom constraint. The key in the constraint refers to an Istio attribute name
```

Comment:
For now my preference is to remove any abstract constraints. I don't see how they make sense for local execution in the proxy?